uefi-raw/src/protocol/nvme.rs

```diff
@@ -30,8 +77,20 @@ pub struct NvmExpressCommand {
     pub cdw15: u32,
 }
 
+newtype_enum! {
+    /// Type of queues an NVMe command can be placed into
+    /// (Which queue a command should be placed into depends on the command)
+    #[derive(Default)]
+    pub enum NvmExpressQueueType: u8  => {
+        /// Admin Submission Queue
+        ADMIN = 0,
+        /// 1) I/O Submission Queue
+        IO = 1,
+    }
+}
+
 /// This structure maps to the NVM Express specification Completion Queue Entry
-#[derive(Debug)]
+#[derive(Debug, Default)]
```

Comment:
nit: why do we need Default here?

Reply:
It's used in `NvmeNamespace::execute_command()`.
That struct is filled by the device, so it's never initialized with any meaningful values from the rust side.